test/boot_test.mjs

```diff
@@ -27,6 +27,12 @@ test("should be able getConfig for a valid path", async (t) => {
   await t.notThrowsAsync(() => getConfig(resolve(__dirname, "../config.mjs")));
 });
 
+test("if getConfig fails on invalid config", async (t) => {
+  await t.throwsAsync(() =>
+    getConfig(resolve(__dirname, "../test/fixtures/falseconfig.mjs"))
```

Comment:
Since this is not throwing, my question is:

- Are the endpoint values optional to configure or are we strictly relying on them? https://github.com/neume-network/core/pull/72/files#diff-826acc8bbab78aeb20092c5e399484062e9895f2056e46234e8730eee3a882b6R9

Reply:
Endpoints are optional. It can be that our crawler makes a request to an endpoint which isn't tracked in the config. Maybe a new NFT has a new tokenURI not seen before. I didn't want the crawler to break in such scenarios so made the endpoint config optional.

It can been seen below that rateLimiter is used only if it has been defined for the specific endpoint.

```js
const { rateLimiter, timeout: timeoutFromConfig } =
  endpointStore.get(origin) ?? {};
if (rateLimiter) {
  await rateLimiter.removeTokens(1);
}

if (options.timeout || timeoutFromConfig) {
  options.signal = AbortSignal.timeout(
    options.timeout ?? timeoutFromConfig
  );
  delete options.timeout;
}
``` 
> From: https://github.com/neume-network/extraction-worker/blob/main/src/api.mjs#L100-L111

Reply:
ok, then my false config fixture is actually not false. It's valid. Let me try testing something else then...